Ensure that there is some form of label for storage locations and they don't overwrite each other

ross-spencer opened the pull request:
In this PR I construct a new data structure to return from 'Job: Retrieve storage locations' this data structure is designed to avoid indexes being overwritten by empty values. The data structure looks as follows:

```
{
    "7f111c13-a54d-49bb-8b75-b3eae94c2a27": {
        "description": "/var/archivematica/AIPsStore/2", 
        "uri": "/api/v2/location/7f111c13-a54d-49bb-8b75-b3eae94c2a27/"
    }, 
    "b2113a69-ecb9-4a4c-b81d-9983de3657b7": {
        "description": "Store AIP in standard Archivematica Directory", 
        "uri": "/api/v2/location/b2113a69-ecb9-4a4c-b81d-9983de3657b7/"
    }, 
    "d13cc6f5-6d8c-4f4f-943e-73a68ad30335": {
        "description": "/var/archivematica/AIPsStore/3", 
        "uri": "/api/v2/location/d13cc6f5-6d8c-4f4f-943e-73a68ad30335/"
    }, 
    "default": {
        "description": "Default Location", 
        "uri": "/api/v2/location/default/AS/"
    }
}
```

Storage locations are required to be displayed, and used, by `Job: Store AIP location`. As described in archivematica/issues#456 if a storage location is blank, or is a duplicate to another description, then the storage locations list will be incomplete - duplicate values overwrite each other. 

In the example above, we would will now see:
```
Default Location
Store AIP in standard Archivematica Directory
/var/archivematica/AIPsStore/2
/var/archivematica/AIPsStore/3
```

Where previously their descriptions being blank would mean we see:
```
Default Location
Store AIP in standard Archivematica Directory

```

**NB.** The last field is simply blank, and to the end-user could be one of either `AIPsStore/2` or `AIPsStore/3`. 

As discussed in `#456` the module `linkTaskManagerGetUserChoiceFromMicroserviceGeneratedList.py` is used only by the Store AIP functionality in Archivematica. 

As such, the change to the way it queries the data structure that is now passed to it, should have very little impact other than to make it more robust than perhaps the dictionary sent to it. 

The structure once consumed is removed from the global `passVar` variable. There seems little reason maintaining it here. Once the user has made a choice a modified version of the structure is sent back and added to `passVar` in `proceedWithChoice(...)`. 

A more simple change has been made to `administration/forms.py` where there was a similar issue. In the processing configuration we use a list to store these values so we would see two blank values - none was overwritten by the other. Here we simply output the path in lieu of something more descriptive. 

Both changes have been tested and show to work well in concert with one-another. 

I ran out of time to develop unit tests for either change unfortunately.

Connected to archivematica/issues#456

jraddaoui reviewed (approved):
Great work @ross-spencer!!

I just fixed a minor issue in a log and changed the `\` syntax :)

jraddaoui commented:
Hi @ablwr,

I've rebased and added a couple of minor fixes. Please, let me know if you think that's okay and I'll merge with squash.

jraddaoui committed:
Avoid SS locations overwrite

Thanks to Ross Spencer:

Previously AIP or DIP storage locations without labels would overwrite
each other in the dict created to process a user choice. Here we
change that structure to something more robust so that this doesn't
happen and the user can see all of their locations with some indicator
of what that location is whether or not it has a description in the
storage service.

Previously an AIP or DIP storage location in the storage service
without a description would present as a blank field in the
administration form. This is corrected to show the relative_path
when a description hasn't been specified. Similar to the behavior
in the ingest workflow.

Squashed commit of the following:

commit 2a115969e575e70aecac1b72d4b006eaef4d6b9d
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Tue Feb 12 20:13:33 2019 +0100

    Cosmetic changes

commit 185e8dd519496e120b9979306727fb99bcc8a3fe
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Tue Feb 12 20:12:34 2019 +0100

    Fix logging

commit 125d007cf63dea5b02eefd55a7b3121d21464d67
Author: Ross Spencer <all.along.the.watchtower2001@gmail.com>
Date:   Fri Feb 8 23:20:41 2019 +0100

    Provide users with more AIP/DIP store description

    Previously an AIP or DIP storage location in the storage service
    without a description would present as a blank field in the
    administration form. This is corrected to show the relative_path
    when a description hasn't been specified. Similar to the behavior
    in the ingest workflow.

commit 3ba964a6fd1ce9d9402164e9ddd380343b99ec27
Author: Ross Spencer <all.along.the.watchtower2001@gmail.com>
Date:   Fri Feb 8 23:10:06 2019 +0100

    Use a JSON structure to pass AIP/DIP store data

    Previously AIP or DIP storage locations without labels would overwrite
    each other in the dict created to process a user choice. Here we
    change that structure to something more robust so that this doesn't
    happen and the user can see all of their locations with some indicator
    of what that location is whether or not it has a description in the
    storage service.

ablwr commented:
LGTM!
